tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md

```diff
@@ -0,0 +1,606 @@
+
+    ```Javascript
+    ...
+      When.onTheMainPage.onFilterBar()
+          .iChangeFilterField("Status", "A")
+          .and.iExecuteSearch();
+      Then.onTheMainPage.onTable()
+          .iCheckRows(1);
+    });
+    ```
+
+    The filter field **Status** is set to value **A** for "Accepted" to load all records with that status. Additionally, another assertion is entered to check that exactly one record is loaded with this filter.
+
+
+    To again load all available data, clear the filter field by calling the same function with an empty string as the value. Reloading the data shows 5 records.
+
+    ```Javascript
+    ...
+      When.onTheMainPage.onFilterBar()
+          .iChangeFilterField("Status", "", true)
+          .and.iExecuteSearch();
+      Then.onTheMainPage.onTable().iCheckRows(5);
+    });
+    ```
+
+3. Execute an action on a selected item in the list report
+
+    Before executing standard or custom bound actions, you can select an item in the list report table by using the test function `iSelectRows()`. As with the check function for rows, you can determine the selection by a condition. The action to be executed will be the deletion of a travel item. Do this by calling the test function `iExecuteDelete()`.
+
+    The following test fragment selects a single row and deletes the travel item by executing the standard delete action.
+
+    ```Javascript
+    ...
+      When.onTheMainPage.onTable()
+          .iSelectRows({ "Customer": "Ryan (594)" })
+          .and.iExecuteDelete();
+    });
+    ```
+    Since these actions are related to the list report table, you use the identifier `onTable()` to access the corresponding functions. You don´t need to provide a parameter for `onTable()`, since there´s only one table available on the list report. For more functions related to the list report table, please have a look at the content of **TableActions** and **TableAssertions** within the API documentation (see step 4 of this tutorial).
+
+    >Note: There are several test functions for pressing and checking standard buttons like Edit, Delete and Save, that are managed by the SAP Fiori elements framework. Such test functions are e.g. iExecuteEdit/iCheckEdit, iExecuteDelete()/iCheckDelete(), etc. The test library provides the functions iExecuteAction() and iCheckAction() to perform action and assertions on custom action buttons. To identify the button you need to provide an identifier as a parameter for the function (see the related API documentation).
+
+
+    Once the journey reaches the delete function it stops and after some time you´ll see an error message in the log area (see screenshot below). The standard delete action shows a dialog which asks the user to confirm the deletion. This dialog also needs to be handled by the OPA test, i.e. you have to confirm the deletion by pressing the **Delete** button.
```

Comment:
missing comma after "function"
missing comma before "which" - Please check all other "whiches" and add missing commas.

Reply:
done